packages/app-builder-lib/src/targets/snap.ts

```diff
@@ -135,8 +135,10 @@ export default class SnapTarget extends Target {
       delete snap.plugs
     } else {
       const archTriplet = archNameToTriplet(arch)
+      // https://github.com/electron-userland/electron-builder/issues/9320
+      const electronDefault = this.isElectronVersionGreaterOrEqualThan("38.0.0") ? "" : "1"
       appDescriptor.environment = {
-        DISABLE_WAYLAND: options.allowNativeWayland ? "" : "1",
+        DISABLE_WAYLAND: options.allowNativeWayland ? "" : electronDefault,
```

Comment:
I get schema validation errors if this is set to an empty string. Ideally the `DISABLE_WAYLAND` variable wouldn't be in this object at all (even as an empty string) if `allowNativeWayland` is true.

Reply:
Hmmm, what happens if you use `"0"` instead of the empty string?